Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -568,6 +575,12 @@ class ArbitraryMipmapDetector
       return {{SRGBToLinear(p[0]), SRGBToLinear(p[1]), SRGBToLinear(p[2]), SRGBToLinear(p[3])}};
     }
 
+    static PixelRGBAu8 SampleLinear(const u8* src, const Shape& src_shape, u32 x, u32 y)
+    {
+      const auto* p = src + (x + y * src_shape.row_length) * 4;
+      return {{p[0], p[1], p[2], p[3]}};
+    }
+
     // Puts a downsampled image in dst. dst must be at least width*height*4
     static void Downsample(const u8* src, const Shape& src_shape, u8* dst, const Shape& dst_shape)
```

Comment:
These are possibly a __restrict__ candidate?

Reply:
I did a grep and I think `__restrict` is valid.